File: KartMayhem.Services/Database/Staze.cs
namespace KartMayhem.Services.Database
{
    public class Staze
    {
        public int Id { get; set; }

        public string NazivStaze { get; set; }

        public string OpisStaze { get; set; }

        public int CijenaPoOsobi { get; set; }

        public double DuzinaStaze { get; set; }
        
        public int BrojKrugova { get; set; }

        public int MaxBrojOsoba { get; set; }

        public int TezinaId { get; set; }

        public Tezine Tezina { get; set; }

        public ICollection<Rezervacije> Rezervacijes { get; set; } = new List<Rezervacije>();

        public ICollection<Rezencije> Rezencijes { get; } = new List<Rezencije>();
    }
}

File: UI/kartmayhem_desktop/lib/Models/staze.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:kartmayhem_desktop/Models/gradovi.dart';
import 'package:kartmayhem_desktop/Models/tezine.dart';

part 'staze.g.dart';

/// dart run build_runner build

@JsonSerializable()
class Staze {
  int? id;
  String? nazivStaze;
  String? opisStaze;
  int? cijenaPoOsobi;
  double? duzinaStaze;
  int? brojKrugova;
  int? maxBrojOsoba;
  int? brojRezervacija;
  String? slika;
  Tezine? tezina;
  Gradovi? gradovi;

  Staze(
      {this.id,
      this.nazivStaze,
      this.opisStaze,
      this.cijenaPoOsobi,
      this.duzinaStaze,
      this.brojKrugova,
      this.maxBrojOsoba,
      this.brojRezervacija,
      this.slika,
      this.tezina,
      this.gradovi});

  factory Staze.fromJson(Map<String, dynamic> json) => _$StazeFromJson(json);
  Map<String, dynamic> toJson() => _$StazeToJson(this);
}
